cg/test/management/subjects.api.spec.js

```diff
@@ -0,0 +1,231 @@
+const { initResources, fetch, closeResources } = require('../utils');
+const { db } = require('../../src/db');
+const { generateClientKey, encryptForStorage, hash } = require('../../src/utils/encryption');
+const { managementJWT } = require('../../src/utils/jwt');
+const { Unauthorized } = require('../../src/utils/errors');
+
+beforeAll(initResources);
+afterAll(closeResources);
+
+describe('List subjects that have given consent', () => {
+  it('should not allow a manager to list subjects without a manager JWT', async () => {
+    //GIVEN AND WHEN
+    const res = await fetch('/api/management/subjects/list', {
+      method: 'GET',
+      headers: {}
+    });
+
+    //THEN
+    expect(res.ok).toBeFalsy();
+    expect(res.status).toBe(Unauthorized.StatusCode);
+    expect(await res.json()).toEqual(
+      expect.objectContaining({
+        error: 'Authorization header not sent'
+      })
+    );
+  });
+
+  it('should not allow a manager to list subjects with an invalid manager JWT', async () => {
+    //GIVEN AND WHEN
+    const res = await fetch('/api/management/subjects/list', {
+      method: 'GET',
+      headers: {
+        Authorization: `invalid_token`
+      }
+    });
+
+    //THEN
+    expect(res.ok).toBeFalsy();
+    expect(res.status).toBe(Unauthorized.StatusCode);
+    expect(await res.json()).toEqual(
+      expect.objectContaining({
+        error: 'Authorization header failed verification'
+      })
+    );
+  });
+
+  it('should not allow an authentic manager to list subjects whose personal data is empty', async () => {
+    //GIVEN
+    const subjectData = {}; // Empty subject's personal data implies that the subject should not be listed
+    const encryptionKey = generateClientKey();
+    const encryptedSubjectData = encryptForStorage(JSON.stringify(subjectData), encryptionKey);
+    const subjectIdHash = hash('subject652848158'); // Random ID to not influence other tests
+
+    await db('subjects').insert({
+      id: subjectIdHash,
+      personal_data: encryptedSubjectData
+    });
+
+    await db('subject_keys').insert({
+      subject_id: subjectIdHash,
+      key: encryptionKey
+    });
+
+    //WHEN
+    const managementToken = await managementJWT.sign({ id: 1 });
+
+    const res = await fetch('/api/management/subjects/list', {
+      method: 'GET',
+      headers: {
+        Authorization: `Bearer ${managementToken}`
+      }
+    });
+
+    //THEN
+    expect(res.ok).toBeTruthy();
+    expect(res.status).toBe(200);
+    expect(await res.json()).toEqual(
+      expect.arrayContaining([])
```

Comment:
I think this will be non-deterministic, since other tests can fill some other subjects. We should expect, that the array does not contain the particular subject, that has been added in this test.